crowbar_framework/app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class RepositoriesController < ApplicationController
+  before_filter :load_registry
```

Comment:
Isn't it clear from what I just said about class variable memoization persisting within each rainbows/puma thread between HTTP requests?  If we don't do it per request, we'll almost certainly end up using out of date data.  And more than once per request doesn't make sense.  So `before_filter` is one way of achieving this, although I'm not sure whether it's the best way...

Another way might be to maintain timestamps for the memoized config, and if we can see newer timestamps any of the repos, empty the cache so that it gets repopulated via JIT loading.  But that might be more effort than it's worth.

Instead of timestamps, maybe there is a way of uniquely identifying each request?  But even if so, this would presumably require special knowledge of the request which only the controller has, preventing us from reaching the ideal scenario where we would avoid the current violation of the principle of least knowledge caused by the controller knowing about the model's cache (which is an implementation detail).  So I doubt there's a perfect solution here.